spec/consensus/light-client.md

```diff
@@ -60,259 +61,368 @@ For the purpose of this lite client specification, we assume that the Tendermint
+3. **Bisection.** If a check according to the trusting period fails, the lite client can try to obtain a header *hp* whose height lies between *h* and *snh* in order to check whether *h* can be used to get trust for *hp*, and *hp* can be used to get trust for *snh*. If this is the case we can trust *snh*; if not, we may continue recursively.
 
 ## How to use it
 
 We consider the following use case:
-the lite client wants to verify a header for some given height _k_. Thus:
-
-- it requests the signed header for height _k_ from a full node
-- it tries to verify this header with the methods described here.
+ the lite client wants to verify a header for some given height *k*. Thus:
+  - it requests the signed header for height *k* from a full node
+  - it tries to verify this header with the methods described here.
 
 This can be used in several settings:
+  - someone tells the lite client that application data that is relevant for it can be read in the block of height *k*.
+  - the lite clients wants the latest state. It asks a full nude for the current height, and uses the response for *k*.
 
-- someone tells the lite client that application data that is relevant for it can be read in the block of height _k_.
-- the lite clients wants the latest state. It asks a full node for the current height, and uses the response for _k_.
 
 ## Details
 
-_Assumptions_
-
-1. _tp < unbonding period_.
-2. _snh.Header.bfttime < now_
-3. _snh.Header.bfttime < h.Header.bfttime+tp_
-4. _trust(h)=true_
+**Observation 1.** If *h.Header.bfttime + tp > now*, we trust the old
+validator set *h.Header.NextV*.
 
-**Observation 1.** If _h.Header.bfttime + tp > now_, we trust the old
-validator set _h.Header.NextV_.
+When we say we trust *h.Header.NextV* we do *not* trust that each individual validator in *h.Header.NextV* is correct, but we only trust the fact that at most 1/3 of them are faulty (more precisely, the faulty ones have at most 1/3 of the total voting power).
 
-When we say we trust _h.Header.NextV_ we do _not_ trust that each individual validator in _h.Header.NextV_ is correct, but we only trust the fact that at most 1/3 of them are faulty (more precisely, the faulty ones have at most 1/3 of the total voting power).
 
 ### Functions
 
-The function _Bisection_ checks whether to trust header _h2_ based on the trusted header _h1_. It does so by calling
-the function _CheckSupport_ in the process of
-bisection/recursion. _CheckSupport_ implements the trusted period method and, for two adjacent headers (in term of heights), it checks uninterrupted sequence of proof.
+The function *CanTrust* checks whether to trust header *h2* based on the trusted header *h1*. It does so by (potentially)
```

Comment:
Why don't we name these vars `trusted_h` and `untrusted_h`. The names would speak for themselves.